src/scripts/main.js

```diff
@@ -1,3 +1,424 @@
 'use strict';
 
-// write code here
+const tableBody = document.querySelector('tbody');
+const tableHead = document.querySelector('thead');
+const rows = [...tableBody.rows];
+
+// Sorting table
+
+let typeOfSort = 'DESC';
+let sortPrevTarget;
+
+function sortTable(sortIndex, array, type) {
+  const text = rows[0].cells[sortIndex].innerText;
+
+  const getSortedRow = (a, b) => {
+    if (!isNaN(Number(text))) {
+      return a.cells[sortIndex].innerText - b.cells[sortIndex].innerText;
+    };
+
+    if (text[0] === '$') {
+      return convertToNumber(a.cells[sortIndex].innerText)
+        - convertToNumber(b.cells[sortIndex].innerText);
+    };
+
+    return a.cells[sortIndex].innerText
+      .localeCompare(b.cells[sortIndex].innerText);
+  };
+
+  const sortedRow = type === 'ASC'
+    ? array.sort((rowA, rowB) => getSortedRow(rowA, rowB))
+    : array.sort((rowA, rowB) => getSortedRow(rowB, rowA));
+
+  tableBody.append(...sortedRow);
+};
+
+tableHead.addEventListener('click', (e) => {
+  const indexForSorting = [...tableHead.rows[0].cells].indexOf(e.target);
+
+  if (typeOfSort === 'DESC' || sortPrevTarget !== e.target) {
+    sortTable(indexForSorting, [...tableBody.rows], 'ASC');
+    typeOfSort = 'ASC';
+  } else {
+    sortTable(indexForSorting, [...tableBody.rows], 'DESC');
+    typeOfSort = 'DESC';
+  };
+
+  sortPrevTarget = e.target;
+});
+
+function convertToNumber(string) {
+  return +string.slice(1).split(',').join('');
+};
+
+// Selected Row
+
+let selectedElement;
+
+tableBody.addEventListener('click', (e) => {
+  const row = e.target.parentElement;
+
+  if (!selectedElement) {
+    row.classList.add('active');
+    selectedElement = row;
+  } else if (selectedElement !== row) {
+    selectedElement.classList.remove('active');
+
+    row.classList.add('active');
+    selectedElement = row;
+  };
+});
+
+// AddForm Script
```

Comment:
ok, but for the future, think about using `insertAdjacentHTML` in a similar situation. it may be shorter